src/core_test/mpas_test_core_halo_exch.F

```diff
@@ -1370,5 +1383,227 @@ subroutine test_core_halo_exch_validate_fields(domain, threadErrs, err)!{{{
 
    end subroutine test_core_halo_exch_validate_fields!}}}
 
+   !***********************************************************************
+   !> \brief   Identify cells which are adjacent to the cells with the provided cellMask/
+   !> \Author Michael Duda
+   !> \details 
+   !>  This function identifies cells which are adjacent to cells with the provided mask.
+   !>  Such cells will be marked with with cellMask+1.
+   !>  Only cells which have no mask are considered.
+   !-----------------------------------------------------------------------
+   function mark_interior_cells(cellMask, sentinelValue, cellsOnCell, nEdgesOnCell) result(nCellsMarked)
+      !< array of masks, where a value of zero means the cell won't be updated by the halo adjoint exchange
+      integer, dimension(:), intent(inout) :: cellMask
+      integer, intent(in) :: sentinelValue !< the value to look for being adjacent to
+      integer, dimension(:,:), intent(in) :: cellsOnCell   !< array containing adjacent cells for each cell
+      integer, dimension(:), intent(in) :: nEdgesOnCell  !< array containing edges for each cell
+
+      integer :: iCell, j, nCellsMarked
+
+      nCellsMarked = 0
+      do iCell = 1, size(cellMask)
+         if (cellMask(iCell) == 0) then
+            do j = 1, nEdgesOnCell(iCell)
+               if (cellMask(cellsOnCell(j, iCell)) == sentinelValue) then
+                  cellMask(iCell) = sentinelValue + 1
+                  nCellsMarked = nCellsMarked + 1
+#ifdef HALO_EXCH_DEBUG_VERBOSE
+                  call mpas_log_write(' mark_interior iCell:$i abuts:$i', &
+                          intArgs = (/iCell, cellsOnCell(j,iCell)/))
+#endif
+                   exit
+               end if
+            end do
+         end if
+      end do
+#ifdef HALO_EXCH_DEBUG_VERBOSE
+      call mpas_log_write(' mark_interior nCellsMarked:$i sentinal:$i', &
+          intArgs=(/nCellsMarked, sentinelValue/))
+#endif
+
+   end function mark_interior_cells
+
+   !***********************************************************************
+   !> \brief   Identify interior and edge cells
+   !> \Author Michael Duda, Jim Wittig
+   !> \details 
+   !>  This function identifies cells which are on the interior, vs cells on the edge
+   !>  (cells with neighbors which aren't in this block).
+   !>  Returns a vector of flags corresponding to the array of cells, where a value of zero indicates
```

Comment:
Does this comment make sense? Is there a better place to document how the number of halo layers impacts which cells get updated during the adjoint halo exchange?